src/eth/primitives/index.rs

```diff
@@ -0,0 +1,27 @@
+use sqlx::database::HasValueRef;
+use sqlx::error::BoxDynError;
+
+use crate::gen_newtype_from;
+
+pub struct Index(i32);
```

Comment:
* Please add a doc explaining it is used in TransactionIndex and LogIndex.
* Use a unsigned integer because we cannot have a negative index.